EssentialsGeoIP/src/com/earth2me/essentials/geoip/EssentialsGeoIPPlayerListener.java

```diff
@@ -175,6 +175,12 @@ private void downloadDatabase() {
                 logger.log(Level.SEVERE, tl("geoIpUrlEmpty"));
                 return;
             }
+            String licenseKey = config.getString("database.license-key", "");
+            if (licenseKey == null || licenseKey.isEmpty()) {
```

Comment:
Given the default when will licenceKey ever be null?

Maybe instead we should remove the default and check for null.

Reply:
Not sure. A few lines up the same code is used to check the URL though, so I just went based off of that. I figure you can never be too careful.

Reply:
Alright let me be more straight forwards.

Given that you defined a default, licenseKey will never be null.